src/transport/CASESession.cpp

```diff
@@ -54,15 +71,145 @@ void CASESession::Clear()
+{
+    CHIP_ERROR error = CHIP_NO_ERROR;
+    CASESessionSerializable serializable;
+    size_t maxlen            = BASE64_ENCODED_LEN(sizeof(serializable));
+    size_t len               = strnlen(Uint8::to_char(input.inner), maxlen);
+    uint16_t deserializedLen = 0;
+
+    VerifyOrExit(len < sizeof(CASESessionSerialized), error = CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrExit(CanCastTo<uint16_t>(len), error = CHIP_ERROR_INVALID_ARGUMENT);
+
+    memset(&serializable, 0, sizeof(serializable));
+    deserializedLen =
+        Base64Decode(Uint8::to_const_char(input.inner), static_cast<uint16_t>(len), Uint8::to_uchar((uint8_t *) &serializable));
+
+    VerifyOrExit(deserializedLen > 0, error = CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrExit(deserializedLen <= sizeof(serializable), error = CHIP_ERROR_INVALID_ARGUMENT);
+
+    error = FromSerializable(serializable);
+
+exit:
+    return error;
+}
+
+CHIP_ERROR CASESession::ToSerializable(CASESessionSerializable & serializable)
+{
+    CHIP_ERROR error = CHIP_NO_ERROR;
+
+    const NodeId peerNodeId = mConnectionState.GetPeerNodeId();
+    VerifyOrExit(CanCastTo<uint16_t>(mI2RR2IKeyLen), error = CHIP_ERROR_INTERNAL);
+    VerifyOrExit(CanCastTo<uint64_t>(mLocalNodeId), error = CHIP_ERROR_INTERNAL);
+    VerifyOrExit(CanCastTo<uint64_t>(peerNodeId), error = CHIP_ERROR_INTERNAL);
+
+    memset(&serializable, 0, sizeof(serializable));
+    serializable.mI2RR2IKeyLen    = static_cast<uint16_t>(mI2RR2IKeyLen);
+    serializable.mPairingComplete = (mPairingComplete) ? 1 : 0;
+    serializable.mLocalNodeId     = mLocalNodeId;
+    serializable.mPeerNodeId      = peerNodeId;
+    serializable.mLocalKeyId      = mConnectionState.GetLocalKeyID();
+    serializable.mPeerKeyId       = mConnectionState.GetPeerKeyID();
+
+    memcpy(serializable.mI2RR2IKey, mI2RR2IKey, mI2RR2IKeyLen);
+
+exit:
+    return error;
+}
+
+CHIP_ERROR CASESession::FromSerializable(const CASESessionSerializable & serializable)
+{
+    CHIP_ERROR error = CHIP_NO_ERROR;
+
+    mPairingComplete = (serializable.mPairingComplete == 1);
+    mI2RR2IKeyLen    = static_cast<size_t>(serializable.mI2RR2IKeyLen);
+
+    VerifyOrExit(mI2RR2IKeyLen <= sizeof(mI2RR2IKey), error = CHIP_ERROR_INVALID_ARGUMENT);
+    memset(mI2RR2IKey, 0, sizeof(mI2RR2IKey));
+    memcpy(mI2RR2IKey, serializable.mI2RR2IKey, mI2RR2IKeyLen);
+
+    mLocalNodeId = serializable.mLocalNodeId;
+    mConnectionState.SetPeerNodeId(serializable.mPeerNodeId);
+    mConnectionState.SetLocalKeyID(serializable.mLocalKeyId);
+    mConnectionState.SetPeerKeyID(serializable.mPeerKeyId);
+
+exit:
+    return error;
+}
+
+CHIP_ERROR CASESession::Init(ChipOperationalCredentialSet * chipOperationalCredentialSet, NodeId myNodeId, uint16_t myKeyId,
+                             SessionEstablishmentDelegate * delegate)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    VerifyOrExit(delegate != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
We can switch to using `ReturnMacros.h` instead of `VerifyOrExit()` and `SuccessOrExit()`. It'll reduce some boilerplate code. It specifically useful when the `exit:` is just returning error code (instead of doing some additional cleanup).

The code will change to `VerifyOrReturnError(delegate != nullptr, CHIP_ERROR_INVALID_ARGUMENT);`

The following code
```
     err = mCommissioningHash.Begin();
     SuccessOrExit(err);
```

will change to 
```
    ReturnErrorOnFailure(mCommissioningHash.Begin());
```

I understand that the current PR is replicating the code structure from `PASESession.cpp`. Some of that code predates `ReturnMacros` and could use a refresh as well.